test/joystick.js

```diff
@@ -0,0 +1,301 @@
+var MockFirmata = require("./util/mock-firmata"),
+  five = require("../lib/johnny-five.js"),
+  sinon = require("sinon"),
+  Board = five.Board,
+  Joystick = five.Joystick;
+
+var io = new MockFirmata();
+var board = new Board({
+  io: io,
+  debug: false,
+  repl: false
+});
+
+
+exports["Joystick -- Analog"] = {
+
+  setUp: function(done) {
+
+    this.clock = sinon.useFakeTimers();
+    this.analogRead = sinon.spy(MockFirmata.prototype, "analogRead");
```

Comment:
When using the instance, I've been experiencing strange behavior that I don't have the time resources to deeply investigate. While trying to figure that out, I discovered that using the prototype eliminated the issue and I've been converting them as I go along since. There is plenty that I dislike about solving it this way and even the explanation, but I don't have the time in the day to spend figuring it out.